java_src/foam/nanos/audit/AuditDAO.java

```diff
@@ -38,10 +40,22 @@ private String formatMessage(FObject currentValue, FObject newValue) {
 
   @Override
   public FObject put(FObject obj) {
+    // TODO: use context-oriented context when available.
     User user = (User) getX().get("user");
     NanoLogger logger = (NanoLogger) getX().get("logger");
     FObject current = this.find(obj);
     logger.info("CHANGE", obj.getClassInfo().getId(), user.getId(), formatMessage(current, obj));
     return super.put(obj);
   }
+
+  @Override
+  public FObject remove(FObject obj) {
+    // TODO: use context-oriented context when available.
+    User user = (User) getX().get("user");
+    NanoLogger logger = (NanoLogger) getX().get("logger");
+    StringBuilder sb = new StringBuilder();
+    new Outputter().output(sb, obj);
+    logger.info("REMOVE", obj.getClassInfo().getId(), user.getId(), sb.toString());
```

Comment:
Just pass 'sb', not 'sb.toString()'. The logger will call toString() on your arguments anyway, and this way, if this log message is filtered out, then you saved the effort.